src/Component/ContentElement/AbstractContentElement.php

```diff
@@ -42,11 +42,11 @@ public function generate(): string
      */
     protected function isVisible(): bool
     {
-        if (TL_MODE !== 'FE' || !BE_USER_LOGGED_IN) {
+        if (TL_MODE === 'FE' && BE_USER_LOGGED_IN) {
```

Comment:
With your proposed the visibility settings (invisible, start, stop) would be recognized in the Contao backend. The constant `BE_USER_LOGGED_IN` is only set in the preview mode. The name is indeed misleading.

I think the correct solution is: `TL_MODE !== 'FE' || BE_USER_LOGGED_IN`

Reply:
That makes sense. I only had the grid extension without backend output. Changed it.

Reply:
That makes sense. I only had the grid extension without backend output for testing. Changed it.